corehq/apps/export/views.py

```diff
@@ -326,157 +359,85 @@ def export_list(self):
-        except TaskFailedError:
-            notify_exception(self.request, "Export download failed",
-                             details={'download_id': download_id})
-            return format_angular_error(
-                _("Download Task Failed to Start. It seems that the server "
-                  "might be under maintenance."),
-                log_error=False,
-            )
-        if context.get('is_ready', False):
-            context.update({
-                'dropbox_url': reverse('dropbox_upload', args=(download_id,)),
-                'download_url': "{}?get_file".format(
-                    reverse('retrieve_download', args=(download_id,))
-                ),
-            })
-        context['is_poll_successful'] = True
-        return context
+@require_POST
+@login_and_domain_required
+def prepare_custom_export(request, domain):
+    """Uses the current exports download framework (with some nasty filters)
+    to return the current download id to POLL for the download status.
+    :return: {
+        'success': True,
+        'download_id': '<some uuid>',
+    }
+    """
+    form_or_case = request.POST.get('form_or_case')
+    sms_export = json.loads(request.POST.get('sms_export'))
+    permissions = ExportsPermissionsManager(form_or_case, domain, request.couch_user)
+    permissions.access_download_export_or_404()
 
-    def _get_download_task(self, in_data):
-        export_filters, export_specs = self._process_filters_and_specs(in_data)
-        export_instances = [self._get_export(self.domain, spec['export_id']) for spec in export_specs]
-        self._check_deid_permissions(export_instances)
-        self._check_export_size(export_instances, export_filters)
-        return get_export_download(
-            export_instances=export_instances,
-            filters=export_filters,
-            filename=self._get_filename(export_instances)
+    try:
+        filter_form_data = json.loads(request.POST.get('form_data'))
+        export_specs = json.loads(request.POST.get('exports'))
+        mobile_user_and_group_slugs = _get_mobile_user_and_group_slugs(
+            filter_form_data[ExpandedMobileWorkerFilter.slug]
         )
-
-    @staticmethod
-    def _get_form_data_and_specs(in_data):
         try:
-            export_specs = in_data['exports']
-            filter_form_data = in_data['form_data']
-        except (KeyError, TypeError):
-            raise ExportAsyncException(
-                _("Request requires a list of exports and filters.")
-            )
+            # Determine export filter
+            form_class = _get_filter_form_class(form_or_case, sms_export)
+            domain_object = Domain.get_by_name(domain)
+            timezone = _get_timezone(domain, request.couch_user)
+            filter_form = form_class(domain_object, timezone, filter_form_data)
+            if not filter_form.is_valid():
+                raise ExportFormValidationException()
```

Comment:
Is this the only place this is raised?  If so, you could remove the `try/except` block and `raise ExportAsyncException` here.  Looks like at the moment this raises, is caught, and then raised again differently.

Reply:
Cleaned up in 0882e62b82c